### DISCUSSION TITLE
Listening for transaction before commit events globally

### DISCUSSION THREAD

**Discussion by erkieh** [micronaut-projects/micronaut-data#2680] [Q&A] [answered]
Hi,

Is it possible to listen transaction before commit events globally, by setting up a listener on startup in Micronaut 4? 

I was able to provide my own DataSourceTransactionManager in Micronaut 3 by extending the default one, but in Micronaut 4 the class is final.

**Comment by dstepanov** [comment, marked as answer]
Not at the moment, but we can add it. Please create an issue.

**Reply by erkieh** [reply]
Thank you for the answers Denis Stepanov! I have created an issue.
https://github.com/micronaut-projects/micronaut-data/issues/2682

**Comment by erkieh** [comment]
What about listening to connection usage to set transaction context in the database?

**Reply by dstepanov** [reply]
In what way? Do you have an idea for an API?

**Reply by erkieh** [reply]
I was thinking if of somehow doing @Around on some java.sql.Connection methods. But since the connections are obtained from datasource I am not sure if there is a way to do it with Micronaut AOP annotations.

**Reply by dstepanov** [reply]
What is your use case? Do you care more about TXs or connections?

**Reply by erkieh** [reply]
I want to do some async post processing after a successful commit so I care more about tx.
